File: ios/Sources/HovrDevicePermissions/location/LocationAccessEvaluator.swift
import CoreLocation

enum LocationAccessEvaluator {
    static func evaluate(
        locationServicesEnabled: Bool,
        authorization: CLAuthorizationStatus
    ) -> PermissionAccessIssue {
        if !locationServicesEnabled {
            return .deviceLocationDisabled
        }
        switch authorization {
        case .restricted:
            return .deviceLocationRestricted
        case .authorizedAlways, .authorizedWhenInUse:
            return .granted
        case .denied:
            return .appLocationDenied
        case .notDetermined:
            return .notDetermined
        @unknown default:
            return .appLocationDenied
        }
    }
}

File: android/hovr_device_permissions/src/main/kotlin/com/hovr/devicepermissions/location/LocationAccessEvaluator.kt
package com.hovr.devicepermissions.location

import com.hovr.devicepermissions.PermissionAccessIssue
import com.hovr.devicepermissions.PermissionStatus

internal object LocationAccessEvaluator {
    fun evaluate(
        locationServicesEnabled: Boolean,
        permissionStatus: PermissionStatus,
    ): PermissionAccessIssue {
        if (!locationServicesEnabled) {
            return PermissionAccessIssue.DEVICE_LOCATION_DISABLED
        }
        return when (permissionStatus) {
            PermissionStatus.GRANTED -> PermissionAccessIssue.GRANTED
            PermissionStatus.NOT_DETERMINED -> PermissionAccessIssue.NOT_DETERMINED
            PermissionStatus.DENIED,
            PermissionStatus.DENIED_PERMANENTLY,
            -> PermissionAccessIssue.APP_LOCATION_DENIED
            else -> PermissionAccessIssue.APP_LOCATION_DENIED
        }
    }
}
